core/src/main/java/org/polypheny/db/catalog/entity/logical/LogicalView.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019-2023 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.catalog.entity.logical;
+
+
+import com.google.common.collect.ImmutableMap;
+import io.activej.serializer.annotations.Deserialize;
+import io.activej.serializer.annotations.Serialize;
+import java.io.Serial;
+import java.util.List;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import lombok.experimental.SuperBuilder;
+import org.polypheny.db.algebra.AbstractAlgNode;
+import org.polypheny.db.algebra.AlgCollation;
+import org.polypheny.db.algebra.AlgNode;
+import org.polypheny.db.algebra.BiAlg;
+import org.polypheny.db.algebra.SingleAlg;
+import org.polypheny.db.algebra.logical.relational.LogicalRelViewScan;
+import org.polypheny.db.catalog.Catalog;
+import org.polypheny.db.catalog.logistic.EntityType;
+import org.polypheny.db.languages.QueryLanguage;
+import org.polypheny.db.plan.AlgCluster;
+import org.polypheny.db.view.ViewManager.ViewVisitor;
+
+@EqualsAndHashCode(callSuper = true)
+@SuperBuilder(toBuilder = true)
+@Value
+@NonFinal
+public class LogicalView extends LogicalTable {
+
+    @Serial
+    private static final long serialVersionUID = -4771308114962700515L;
+
+    @Serialize
+    public ImmutableMap<Long, List<Long>> underlyingTables;
+    @Serialize
+    public QueryLanguage language;
+    @Serialize
+    public AlgCollation algCollation;
+    @Serialize
+    public String query;
```

Comment:
I think it would be better to store the query in its algebra representation as soon as we have the textual algebra representation. This allows names to change since we then have the ids in the query. We also do not need to parse every time.

Reply:
But this would require to serialize the AlgNode for persistance purposes which is as of now, not available in a stable manner. This will hopefully change soon.